Samples/Samples.Cards/Program.cs

```diff
@@ -165,351 +193,495 @@ public async Task<ActionResponse> OnCardAction([Context] AdaptiveCards.ActionAct
-                                    }
-                                ],
-                                ""width"": ""stretch""
-                            }
-                        ]
-                    },
-                    {
-                        ""type"": ""TextBlock"",
-                        ""text"": ""This card was created from JSON deserialization!"",
-                        ""wrap"": true,
-                        ""color"": ""good"",
-                        ""spacing"": ""medium""
-                    }
-                ],
-                ""actions"": [
-                    {
-                        ""type"": ""Action.Execute"",
-                        ""title"": ""Test JSON Action"",
-                        ""data"": {
-                            ""Value"": {
-                                ""action"": ""test_json""
-                            }
-                        },
-                        ""associatedInputs"": ""auto""
-                    }
-                ],
-                ""version"": ""1.5"",
-                ""schema"": ""http://adaptivecards.io/schemas/adaptive-card.json""
-            }";
+            log.Info($"[MESSAGE] Received: {SanitizeForLog(activity.Text)}");
+            log.Info($"[MESSAGE] From: {SanitizeForLog(activity.From?.Name ?? "unknown")}");
+
+            var text = activity.Text?.ToLowerInvariant() ?? "";
+
+            if (text.Contains("card"))
+            {
+                log.Info("[CARD] Basic card requested");
+                var card = CreateBasicAdaptiveCard();
+                await client.Send(card);
+            }
+            else if (text.Contains("profile"))
+            {
+                log.Info("[PROFILE] Profile card requested");
+                var card = CreateProfileCard();
+                await client.Send(card);
+            }
+            else if (text.Contains("validation"))
+            {
+                log.Info("[VALIDATION] Validation card requested");
+                var card = CreateProfileCardWithValidation();
+                await client.Send(card);
+            }
+            else if (text.Contains("feedback"))
+            {
+                log.Info("[FEEDBACK] Feedback card requested");
+                var card = CreateFeedbackCard();
+                await client.Send(card);
+            }
+            else if (text.Contains("form"))
+            {
+                log.Info("[FORM] Task form card requested");
+                var card = CreateTaskFormCard();
+                await client.Send(card);
+            }
+            else if (text.Contains("json"))
+            {
+                log.Info("[JSON] JSON deserialization card requested");
+                var card = CreateCardFromJson();
+                await client.Send(card);
+            }
+            else if (text.Contains("reply"))
+            {
+                await client.Send("Hello! How can I assist you today?");
```

Comment:
nit.. this is the typical message generated by GTP.. what is the intent of the reply command? maybe remove it?